client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/coverage/methodreplacement/thirdpartyclasses/DocumentClassReplacement.java

```diff
@@ -0,0 +1,56 @@
+package org.evomaster.client.java.instrumentation.coverage.methodreplacement.thirdpartyclasses;
+
+import org.evomaster.client.java.instrumentation.coverage.methodreplacement.Replacement;
+import org.evomaster.client.java.instrumentation.coverage.methodreplacement.ThirdPartyMethodReplacementClass;
+import org.evomaster.client.java.instrumentation.coverage.methodreplacement.UsageFilter;
+import org.evomaster.client.java.instrumentation.object.JsonTaint;
+import org.evomaster.client.java.instrumentation.shared.ReplacementCategory;
+import org.evomaster.client.java.instrumentation.shared.ReplacementType;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+public class DocumentClassReplacement extends ThirdPartyMethodReplacementClass {
+
+    public static final String ORG_BSON_DOCUMENT = "org.bson.Document";
+
+    @Override
+    protected String getNameOfThirdPartyTargetClass() {
+        return ORG_BSON_DOCUMENT;
+    }
+
+    @Replacement(replacingStatic = true,
+            type = ReplacementType.TRACKER,
+            id = "Document_parse_String",
+            usageFilter = UsageFilter.ANY,
+            category = ReplacementCategory.MONGO,
+            castTo = "org.bson.Document")
+    public static Object parse(String json) {
+        // Load the Document class
+        Class<?> documentClass;
+        try {
+            documentClass = Class.forName("org.bson.Document");
```

Comment:
in instrumentation, a call to `Class.forName` is extremely tricky. for example, the classloader of `Class` and of Mongo driver could be different (eg in a SpringBoot application run with external driver).
have a look at all the shenanigans had to go through in `ThirdPartyMethodReplacementClass.getOriginalConstructor`.
in other words, this might work in E2E tests, but fail to work in the experiments.
likely would need to use `UnitsInfoRecorder.getInstance().getClassLoaders`

Reply:
I pushed a way of handling class loaders for static methods in ThirdPartyMethodReplacementClass. Please have a look at getOriginalStaticMethod(), specially how the callerName is obtained.